add tiup no-sudo doc (#20179)
pingcap/docs: tiup/tiup-cluster-topology-reference.md

- `group`: The user group to which a user belongs. It is specified when the user is created. The value defaults to that of the `<user>` field. If the specified group does not exist, it is automatically created.

> - `systemd_mode`: The systemd mode used on the target machine during cluster deployment. The default value is `system`. If it is set to `user`, it means that sudo permissions are not used on the target machine, that is, the TiUP no-sudo mode is used.

Reviewer: ```suggestion
- `systemd_mode`: The `systemd` mode used on the target machine during cluster deployment. The default value is `system`. If you set it to `user`, it means that sudo permissions are not used on the target machine, that is, the [TiUP no-sudo mode](/tiup/tiup-cluster-no-sudo-mode.md) is used.
```

> - `systemd_mode`: The `systemd` mode used on the target machine during cluster deployment. The default value is `system`. If you set it to `user`, it means that sudo permissions are not used on the target machine, that is, the [TiUP no-sudo mode](/tiup/tiup-cluster-no-sudo-mode.md) is used.

Reviewer: ```suggestion
- `systemd_mode`: The `systemd` mode used on the target machine during cluster deployment. The default value is `system`. If set to `user`, sudo permissions are not required on the target machine, meaning [TiUP no-sudo mode](/tiup/tiup-cluster-no-sudo-mode.md) is used.
```
